src/main/java/org/springframework/analytics/rest/controller/AggregateCounterController.java

```diff
@@ -107,7 +113,7 @@ protected void delete(@PathVariable("name") String name) {
 		List<String> names = new ArrayList<>(repository.list());
 		long count = names.size();
 		long pageEnd = Math.min(count, pageable.getOffset() + pageable.getPageSize());
-		Page aggregateCounterPage = new PageImpl<>(names.subList(pageable.getOffset(), (int) pageEnd), pageable, names.size());
+		Page aggregateCounterPage = new PageImpl<>(names.subList((int)pageable.getOffset(), (int) pageEnd), pageable, names.size());
```

Comment:
though unlikely to be encountered, we can use `import static java.lang.Math.toIntExact;` to 'safely' convert from long to int.